bundles/aliyun/src/main/java/org/apache/gravitino/oss/fs/OSSCredentialProvider.java

```diff
@@ -0,0 +1,154 @@
+
+import com.aliyun.oss.common.auth.BasicCredentials;
+import com.aliyun.oss.common.auth.Credentials;
+import com.aliyun.oss.common.auth.CredentialsProvider;
+import com.aliyun.oss.common.auth.DefaultCredentials;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.client.GravitinoClient;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.credential.OSSSecretKeyCredential;
+import org.apache.gravitino.credential.OSSTokenCredential;
+import org.apache.gravitino.file.Fileset;
+import org.apache.gravitino.file.FilesetCatalog;
+import org.apache.gravitino.filesystem.common.GravitinoVirtualFileSystemConfiguration;
+import org.apache.gravitino.filesystem.common.GravitinoVirtualFileSystemUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.aliyun.oss.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OSSCredentialProvider implements CredentialsProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OSSCredentialProvider.class);
+  private Credentials basicCredentials;
+  private final String filesetIdentifier;
+  private final GravitinoClient client;
+  private final Configuration configuration;
+
+  private long expirationTime = Long.MAX_VALUE;
+  private static final double EXPIRATION_TIME_FACTOR = 0.9D;
+
+  public OSSCredentialProvider(URI uri, Configuration conf) {
+    this.filesetIdentifier =
+        conf.get(GravitinoVirtualFileSystemConfiguration.GVFS_FILESET_IDENTIFIER);
+    this.client = GravitinoVirtualFileSystemUtils.createClient(conf);
+    this.configuration = conf;
+  }
+
+  @Override
+  public void setCredentials(Credentials credentials) {}
+
+  @Override
+  public Credentials getCredentials() {
+    // If the credentials are null or about to expire, refresh the credentials.
+    if (basicCredentials == null || System.currentTimeMillis() >= expirationTime) {
+      synchronized (this) {
+        refresh();
+      }
+    }
+
+    return basicCredentials;
+  }
+
+  private void refresh() {
+    String[] idents = filesetIdentifier.split("\\.");
+    String catalog = idents[1];
+
+    FilesetCatalog filesetCatalog = client.loadCatalog(catalog).asFilesetCatalog();
+
+    Fileset fileset = filesetCatalog.loadFileset(NameIdentifier.of(idents[2], idents[3]));
+    Credential[] credentials = fileset.supportsCredentials().getCredentials();
+    Optional<Credential> optionalCredential = getCredential(credentials);
+
+    if (!optionalCredential.isPresent()) {
+      LOG.warn("No credential found for fileset: {}, try to use static AKSK", filesetIdentifier);
+      expirationTime = Long.MAX_VALUE;
+      this.basicCredentials =
+          new DefaultCredentials(
+              configuration.get(Constants.ACCESS_KEY_ID),
+              configuration.get(Constants.ACCESS_KEY_SECRET));
+      return;
```

Comment:
What if user didn't configure the AKSK? Should they still configure the AKSK when credential vending is enabled?

Reply:
If we can't get any crendential from Gravitino server(dynamic or static), we will try using AKSK. If ASKS has not been set, errors will occur.

Reply:
Should they still configure the AKSK when credential vending is enabled?

I prefer to keep the way that only support AKSK in version 0.7.0

Reply:
Do you need a flag to enable credential vending on the client side like Iceberg client?

Reply:
Have you added an option in the Icebert client to use credentials or not? I want to make credentials transparent to client, so I'm hesitant to add this option in the client side. @jerryshao do you have any comments on this point?

Reply:
The iceberg community added an option in the Icebert client to control whether enable credential vending.

Reply:
> The iceberg community added an option in the Icebert client to control whether enable credential vending.

GVFS supports multiple storage at the same time and aims to shield the difference. I'm afraid the case is quite different from that in Iceberg rest server.